### DISCUSSION TITLE
Lesson 6: Brownie Fund Me - 'WARNING: Unable to compile smartcontractkit/chainlink-brownie-contracts@1.1.1 due to a NamespaceCollision'

### DISCUSSION THREAD

**Discussion by semsion** [smartcontractkit/full-blockchain-solidity-course-py#304] [Q&A] [answered]
Ok, so - at 5h10m58s of Lesson 6: Brownie Fund Me https://youtu.be/M576WGiDBdQ?t=18658 - I'm receiving the following warning:

```
$ ➜  brownie_fund_me brownie compile
Brownie v1.17.0 - Python development framework for Ethereum

865kiB [00:00, 3.05MiB/s]
WARNING: Unable to compile smartcontractkit/chainlink-brownie-contracts@1.1.1 due to a NamespaceCollision - you may still be able to import sources from the package, but will be unable to load the package directly.

Compiling contracts...
  Solc version: 0.6.12
  Optimizer: Enabled  Runs: 200
  EVM Version: Istanbul
Generating build data...
 - smartcontractkit/chainlink-brownie-contracts@1.1.1/AggregatorV3Interface
 - smartcontractkit/chainlink-brownie-contracts@1.1.1/SafeMathChainlink
 - FundMe

Project has been compiled. Build artifacts saved at <pathtohome>/freeCodeCamp_solidity_blockchain_smart_contract_course__python_version/brownie_fund_me/build/contracts

```

The project did actually compile though, and the relevant .json files were populated at build/contracts/dependencies/smartcontractkit/chainlink-brownie-contracts@1.1.1 - so hopefully I don't need to worry too much about this... 

There appears to a similar issue here: https://github.com/eth-brownie/brownie/issues/729 

Has anyone else come across this?

**Comment by cromewar** [comment, marked as answer]
I also got his warning all the time, as is mentioned on the thread you linked it seems brownie does not allows to have contract and interfaces  with the same name. 

I have a hypothesis and I think this happens because we copy-paste the contract manually on our project folder, as we do on the course with `MockV3Aggregator`, so it is also on chainlink's github and brownie warns us saying that we can't use both, so the `MockV3Aggregator` from github is not going to be compiled because we are using a local version.

@PatrickAlphaC  do you have any thoughts on this?

**Reply by PatrickAlphaC** [reply]
Yeah, basically it's finding multiple versions of the same package. If you delete your artifacts it'll get rid of the error.

@semsion can I actually ask you to make this an issue on the brownie repo? Would be good to get some visibility there.

**Reply by semsion** [reply]
Thanks guys - new issue created here: https://github.com/eth-brownie/brownie/issues/1326

@PatrickAlphaC which artefacts are you referring to here?

**Reply by PatrickAlphaC** [reply]
Er sorry... mixing up hardhat and brownie 😅 I wish everything was brownie...

I meant to say the `build` folder.

**Comment by deFaup** [comment]
Interesting, I'm also facing the issue and even if the compilation was going through I couldn't deploy the contract on rinkeby or a local chain. I removed it completely and now I get the error but I can deploy the contract without error.

**Reply by cromewar** [reply]
Hello @deFaup Could you please post here a screenshot of your problem?

**Reply by deFaup** [reply]
Hey, I left a detailed comment here
https://github.com/eth-brownie/brownie/issues/1326

**Reply by cromewar** [reply]
I recommend you to uninstall eth-brownie and delete the `.brownie` folder completely, it is a hidden folder on your home path. This should refresh all the packages and the conflicts between 1.1.1 and 0.2.2... then  reinstall brownie and try again. I hope this might help.

**Comment by matsen01** [comment]
I am facing the same issue. Did you find a solution?
Uninstalling and reinstalling did not solve the problem

**Reply by aviralgoel** [reply]
"If you delete your build folder, it'll get rid of the error." -Patrick

This worked for me.
